src/api/Authentication/DatabaseAuthenticationClaimsTranfomration.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Microsoft.AspNetCore.Authentication;
+using System.Security.Claims;
+
+namespace BDMS.Authentication;
+
+public class DatabaseAuthenticationClaimsTranfomration : IClaimsTransformation
+{
+    private readonly BdmsContext dbContext;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DatabaseAuthenticationClaimsTranfomration"/> class.
+    /// </summary>
+    /// <param name="dbContext">The EF database context containing data for the BDMS application.</param>
+    /// <exception cref="ArgumentNullException">If <paramref name="dbContext"/> is <c>null</c>.</exception>
+    public DatabaseAuthenticationClaimsTranfomration(BdmsContext dbContext)
+    {
+        this.dbContext = dbContext ?? throw new ArgumentNullException(nameof(dbContext));
+    }
+
+    /// <inheritdoc/>
+    public async Task<ClaimsPrincipal> TransformAsync(ClaimsPrincipal principal)
+    {
+        var userId = principal.Claims.FirstOrDefault(c => c.Type == ClaimTypes.NameIdentifier);
+        var authenticatedUser = userId is not null ? dbContext.Users.FirstOrDefault(u => u.Name == userId.Value) : null;
+
+        if (authenticatedUser == null || authenticatedUser.IsDisabled)
```

Comment:
Die Funktionalität für das Deaktivieren und Löschen der Benutzer im BDMS bleibt für den Moment bestehen, da wir den Workflow im Moment noch zu wenig gut kennen und nicht wissen wie und welche Claims/Grants vom eAIM via Cognito transportiert werden.

![image](https://github.com/geoadmin/suite-bdms/assets/1365121/eae58227-3142-4fa8-93cc-9fbbb3dc4c16)

Deshalb müssen wir an dieser Stelle sicherstellen, dass deaktivierte Benutzer nicht einloggen können. Wissen wir was passiert, wenn wir hier `principal` zurückgeben? Braucht es für das ausloggen allenfalls wieder einen einen `AuthenticationHandler` oder geht das auch mit der `ClaimsTransformation` Middleware?

Reply:
Aktuell kann sich ein nur in BDMS deaktivierter (oder neuer ohne Rollen) zwar anmelden, jedoch werden für ihn keine Daten zugänglich sein. Bedeutet aktuell, dass das UI damit noch nicht umgehen kann & darüber keine Information bereitstellt. Das Handling der neuen Edgecases wird über einzelne kleinere Tasks in #833 umgesetzt.